edfi-repo-auditor/edfi_repo_auditor/github_client.py

```diff
@@ -0,0 +1,201 @@
+
+        query = REPOSITORIES_TEMPLATE.replace(ORG_TOKEN, owner)
+        body = self._execute_graphql(f"repositories for {owner}", query)
+
+        total_repos = body["data"]["organization"]["repositories"]["totalCount"]
+        if total_repos > 100:
+            logger.warning(f"There are over 100 repos in {owner}. Update the query to handle pagination")
+
+        df = pd.DataFrame(body["data"]["organization"]["repositories"]["nodes"])
+        return df["name"].to_list()
+
+    def get_actions(self, owner: str, repository: str) -> dict:
+        if len(owner.strip()) == 0:
+            raise ValueError("owner cannot be blank")
+        if len(repository.strip()) == 0:
+            raise ValueError("repository cannot be blank")
+
+        actions = self._execute_api_call(
+            f"Getting actions for {owner}/{repository}", "GET", f"{API_URL}/repos/{owner}/{repository}/actions/workflows"
+        )
+        return actions
+
+    def get_repository_information(self, owner: str, repository: str) -> dict:
+        if len(owner.strip()) == 0:
+            raise ValueError("owner cannot be blank")
+        if len(repository.strip()) == 0:
+            raise ValueError("repository cannot be blank")
+
+        query = REPOSITORY_INFORMATION_TEMPLATE.replace(ORG_TOKEN, owner).replace(
+            REPO_TOKEN, repository
+        )
+
+        body = self._execute_graphql(
+            f"protection rules for {owner}/{repository}", query
+        )
+
+        return body["data"]["repository"]
+
+    def has_dependabot_enabled(self, owner: str, repository: str) -> bool:
+        if len(owner.strip()) == 0:
+            raise ValueError("owner cannot be blank")
+        if len(repository.strip()) == 0:
+            raise ValueError("repository cannot be blank")
+
+        has_dependabot = False
+        try:
+            dependabot = self._execute_api_call(
+                f"Getting actions for {owner}/{repository}", "GET", f"{API_URL}/repos/{owner}/{repository}/vulnerability-alerts"
+            )
+            has_dependabot = dependabot["status_code"] == requests.codes.no_content
+        except RuntimeError:
+            has_dependabot = False
+
+        return has_dependabot
+
+    def get_file_content(self, owner: str, repository: str, path: str) -> str:
+        if len(owner.strip()) == 0:
+            raise ValueError("owner cannot be blank")
+        if len(repository.strip()) == 0:
+            raise ValueError("repository cannot be blank")
+        if len(path.strip()) == 0:
+            raise ValueError("path cannot be blank")
+
+        file_result = None
+        try:
+            file_result = self._execute_api_call(
+                f"Getting file {path} for {owner}/{repository}", "GET", f"{API_URL}/repos/{owner}/{repository}/contents/{path}"
+            )
+        except RuntimeError:
+            logger.warn(f"Unable to get file {path}. Verify that the file is available. Skipping")
```

Comment:
Since this situation is a normal part of the reporting process, let's skip the warning. The output report will tell us that the file does not exist, and that should be sufficient.